src/main/java/org/broadinstitute/hellbender/tools/spark/RevertSamSpark.java

```diff
@@ -0,0 +1,706 @@
+    public boolean dontRemoveAlignmentInformation = false;
+
+    public static final String ATTRIBUTE_TO_CLEAR_ARG = "attributes-to-clear";
+    @Argument(fullName = ATTRIBUTE_TO_CLEAR_ARG,doc = "When removing alignment information, the set of optional tags to remove.", optional = true)
+    public List<String> attributesToClear = new ArrayList<String>();
+
+    public static final String REMOVE_DEFAULT_ATTRIBUTE_TO_CLEAR_ARG = "remove-default-attributes-to-clear";
+    @Argument(fullName = REMOVE_DEFAULT_ATTRIBUTE_TO_CLEAR_ARG,doc = "When removing alignment information, the set of optional tags to remove.", optional = true)
+    public boolean removeDefaults = false;
+
+    public static List<String> DEFAULT_ATTRIBUTES_TO_CLEAR = new ArrayList<String>() {{
+        add(SAMTag.NM.name());
+        add(SAMTag.UQ.name());
+        add(SAMTag.PG.name());
+        add(SAMTag.MD.name());
+        add(SAMTag.MQ.name());
+        add(SAMTag.SA.name()); // Supplementary alignment metadata
+        add(SAMTag.MC.name()); // Mate Cigar
+        add(SAMTag.AS.name());
+    }};
+
+    public static final String SAMPLE_ALIAS_ARG = "sample-alias";
+    @Argument(fullName = SAMPLE_ALIAS_ARG, doc = "The sample alias to use in the reverted output file.  This will override the existing " +
+            "sample alias in the file and is used only if all the read groups in the input file have the " +
+            "same sample alias.", shortName = StandardArgumentDefinitions.SAMPLE_ALIAS_SHORT_NAME, optional = true)
+    public String sampleAlias;
+
+    public static final String LIBRARY_NAME_ARG = "library-name";
+    @Argument(fullName = LIBRARY_NAME_ARG, doc = "The library name to use in the reverted output file.  This will override the existing " +
+            "sample alias in the file and is used only if all the read groups in the input file have the " +
+            "same library name.", optional = true)
+    public String libraryName;
+
+    @Override
+    public List<ReadFilter> getDefaultReadFilters() {
+        return Collections.singletonList(ReadFilterLibrary.ALLOW_ALL_READS);
+    }
+
+    /**
+     * Enforce that output ordering is queryname when sanitization is turned on since it requires a queryname sort.
+     */
+    @Override
+    protected String[] customCommandLineValidation() {
+        final List<String> errors = new ArrayList<>();
+        RevertSamSpark.ValidationUtil.validateOutputParams(outputByReadGroup, output, outputMap, errors);
+
+        if (!sanitize && keepFirstDuplicate) errors.add("'keepFirstDuplicate' cannot be used without 'sanitize'");
+
+        if (!errors.isEmpty()) {
+            return errors.toArray(new String[errors.size()]);
+        }
+        return null;
+    }
+
+    @Override
+    protected void runTool(JavaSparkContext ctx) {
+        Broadcast<SAMFileHeader> headerb = ctx.broadcast(getHeaderForReads());
```

Comment:
It's not clear when to use this and when to use inHeader... It's also confusing because I believe modifying inHeader also modifies this one since they're both just aliases of the same underlying header object.  I don't know how that interacts with the broadcast, but my guess is badly...